cmd/shortener/main.go

```diff
@@ -27,11 +30,33 @@ func initLogger(cfg *config.Config) error {
 }
 
 func initStorage(cfg *config.Config) (*storage.Repository, error) {
-	st, err := storage.NewRepository(cfg.FileStoragePath)
-	if err != nil {
-		return nil, err
+	var st t.Storage
+
+	if cfg.DatabaseDSN != "" {
+		st, err := storage.NewPostgresStorage(cfg.DatabaseDSN)
+		if err != nil {
+			return nil, fmt.Errorf("could not connect to postgres: %w", err)
+		}
+		return storage.NewRepository(st)
+
+	} else if cfg.FileStoragePath != "" {
+		st, err := storage.NewFileStorage(cfg.FileStoragePath)
+		if err != nil {
+			return nil, fmt.Errorf("could not open file %s: %w", cfg.FileStoragePath, err)
+		}
+		return storage.NewRepository(st)
+
+	} else {
+		st = storage.NewMemoryStorage()
+	}
+
+	return storage.NewRepository(st)
```

Comment:
Мне кажется стоит единообразно подойти к возврату значения. Либо наоборот, инициализировать в каждой ветке `st`  и в конце делать `return  storage.NewRepository(st)`.
```suggestion
	if cfg.DatabaseDSN != "" {
		st, err := storage.NewPostgresStorage(cfg.DatabaseDSN)
		if err != nil {
			return nil, fmt.Errorf("could not connect to postgres: %w", err)
		}
		return storage.NewRepository(st)

	} else if cfg.FileStoragePath != "" {
		st, err := storage.NewFileStorage(cfg.FileStoragePath)
		if err != nil {
			return nil, fmt.Errorf("could not open file %s: %w", cfg.FileStoragePath, err)
		}
		return storage.NewRepository(st)

	} else {
		return storage.NewRepository(storage.NewMemoryStorage())
	}
```